assets/model_monitoring/components/src/shared_utilities/df_utils.py

```diff
@@ -3,32 +3,51 @@
 
 """This file contains additional utilities that are applicable to dataframe."""
 import pyspark.sql as pyspark_sql
+import pandas as pd
 
 
-def get_numerical_columns(column_dtype_map: dict) -> list:
+def get_numerical_columns(column_dtype_map: dict, baseline_df) -> list:
     """Get numerical columns from all columns."""
     # NOTE: byte, short, int, long are not included in the list because they are ambiguous with categorical columns.
     # They should be added to the list based on some heuristics or user preference.
     numerical_columns = [
         column
         for column in column_dtype_map
-        if column_dtype_map[column] in ["float", "double", "decimal"]
+        if column_dtype_map[column] in ["float", "double", "decimal"] or column_dtype_map[column] == "int" and is_numerical(pd.Series(baseline_df[column]))
     ]
     return numerical_columns
 
 
-def get_categorical_columns(column_dtype_map: dict) -> list:
+def get_categorical_columns(column_dtype_map: dict, baseline_df) -> list:
     """Get categorical columns from all columns."""
     # NOTE: byte, short, int, long are not included in the list because they are ambiguous with numerical columns.
     # They should be added to the list based on some heuristics or user preference.
     categorical_columns = [
         column
         for column in column_dtype_map
-        if column_dtype_map[column] in ["string", "bool"]
+        if column_dtype_map[column] in ["string", "bool"] or column_dtype_map[column] == "int" and is_categorical(pd.Series(baseline_df[column]))
     ]
     return categorical_columns
 
 
+def get_distinct_ratio(column):
+    distinct_values = column.nunique()
+    total_values = column.size
+    return distinct_values / total_values
+    
+def is_numerical(column):
+    if pd.api.types.is_numeric_dtype(column):
+        distinct_value_ratio = get_distinct_ratio(column)
```

Comment:
So you changed the implementation, the previous one is not correct?

Reply:
no it wasnt